x/bep3/keeper/swap_test.go

```diff
@@ -2,45 +2,481 @@ package keeper_test
+			args{
+				randomNumberHash:    suite.randomNumberHashes[0],
+				timestamp:           suite.timestamps[0],
+				heightSpan:          int64(360),
+				sender:              suite.addrs[0],
+				recipient:           suite.addrs[1],
+				senderOtherChain:    binanceAddrs[0].String(),
+				recipientOtherChain: binanceAddrs[1].String(),
+				coins:               cs(c("bnb", 50000)),
+				expectedIncome:      "50000bnb",
+			},
+			true,
+		},
+		{
+			"unsupported asset",
+			currentTmTime,
+			args{
+				randomNumberHash:    suite.randomNumberHashes[1],
+				timestamp:           suite.timestamps[1],
+				heightSpan:          int64(360),
+				sender:              suite.addrs[1],
+				recipient:           suite.addrs[2],
+				senderOtherChain:    binanceAddrs[0].String(),
+				recipientOtherChain: binanceAddrs[1].String(),
+				coins:               cs(c("xyz", 50000)),
+				expectedIncome:      "50000xyz",
+			},
+			false,
+		},
+		{
+			"past timestamp",
+			currentTmTime,
+			args{
+				randomNumberHash:    suite.randomNumberHashes[2],
+				timestamp:           suite.timestamps[2] - 2000,
+				heightSpan:          int64(360),
+				sender:              suite.addrs[2],
+				recipient:           suite.addrs[3],
+				senderOtherChain:    binanceAddrs[0].String(),
+				recipientOtherChain: binanceAddrs[1].String(),
+				coins:               cs(c("bnb", 50000)),
+				expectedIncome:      "50000bnb",
+			},
+			false,
+		},
+		{
+			"future timestamp",
+			currentTmTime,
+			args{
+				randomNumberHash:    suite.randomNumberHashes[3],
+				timestamp:           suite.timestamps[3] + 5000,
+				heightSpan:          int64(360),
+				sender:              suite.addrs[3],
+				recipient:           suite.addrs[4],
+				senderOtherChain:    binanceAddrs[0].String(),
+				recipientOtherChain: binanceAddrs[1].String(),
+				coins:               cs(c("bnb", 50000)),
+				expectedIncome:      "50000bnb",
+			},
+			false,
+		},
+		{
+			"small height span",
+			currentTmTime,
+			args{
+				randomNumberHash:    suite.randomNumberHashes[4],
+				timestamp:           suite.timestamps[4],
+				heightSpan:          int64(5),
+				sender:              suite.addrs[4],
+				recipient:           suite.addrs[5],
+				senderOtherChain:    binanceAddrs[0].String(),
+				recipientOtherChain: binanceAddrs[1].String(),
+				coins:               cs(c("bnb", 50000)),
+				expectedIncome:      "50000bnb",
+			},
+			false,
+		},
+		{
+			"big height span",
+			currentTmTime,
+			args{
+				randomNumberHash:    suite.randomNumberHashes[5],
+				timestamp:           suite.timestamps[5],
+				heightSpan:          int64(1000000),
+				sender:              suite.addrs[5],
+				recipient:           suite.addrs[6],
+				senderOtherChain:    binanceAddrs[0].String(),
+				recipientOtherChain: binanceAddrs[1].String(),
+				coins:               cs(c("bnb", 50000)),
+				expectedIncome:      "50000bnb",
+			},
+			false,
+		},
+		{
+			"zero amount",
+			currentTmTime,
+			args{
```

Comment:
I would add a case that attempts to create a duplicate swap. You'll have to add a `shouldBeFound` boolean to the top level struct and all for `expectPass=false` tests to have `shouldBeFound=true`

Reply:
added test case in https://github.com/Kava-Labs/kava/pull/395/commits/ad97447a680677cde6630056da816a9d76bdfc25